published/external/xdpapi.h

```diff
@@ -474,6 +486,61 @@ XdpOpenApi(
     return _Status;
 }
 
+#define _POLL_INTERVAL_MS 10
+
+inline
+_IRQL_requires_(PASSIVE_LEVEL)
+NTSTATUS
+XdpOpenApi(
+    _In_ UINT32 _XdpApiVersion,
+    _In_opt_ VOID *_ClientContext,
+    _In_opt_ XDP_API_ATTACH_FN *_ClientAttach,
+    _In_opt_ XDP_API_DETACH_FN *_ClientDetach,
+    _In_ const XDP_API_CLIENT_DISPATCH *_XdpApiClientDispatch,
+    _In_ const INT64 _TimeoutMs,
+    _Out_ XDP_API_CLIENT *_ApiContext,
+    _Out_ const XDP_API_PROVIDER_DISPATCH **_XdpApiProviderDispatch,
+    _Out_ const XDP_API_PROVIDER_BINDING_CONTEXT **_XdpApiProviderContext
+    )
+{
+    NTSTATUS Status;
+    INT64 TimeoutMs = MAXLONGLONG; // forever
```

Comment:
this isn't precisely "forever" because the timeout will gradually be reduced by the code below. it is a very long time, though.